sgt-back-end/index.js

```diff
@@ -0,0 +1,136 @@
+const express = require('express');
+const app = express();
+const pg = require('pg');
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:lfz@localhost/studentGradeTable'
+});
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+    select *
+      from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      res.status(200).json(grades);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.use(express.json());
+
+app.post('/api/grades', (req, res) => {
+  const name = req.body.name;
+  const course = req.body.course;
+  const score = Number(req.body.score);
+  const values = [name, course, score];
+
+  if (score < 0 || score > 100 || !Number.isInteger(score)) {
+    res.status(404).json(
+      { error: 'Must include a score as an interger between 1 - 100' });
+  } else if (!name) {
+    res.status(404).json(
+      { error: 'Must include a name' });
+  } else if (!course) {
+    res.status(404).json(
+      { error: 'Must include a course' });
+  }
```

Comment:
If you don't include `return` statements or put your database query in an `else` block, your server will still try to query the database even if it shouldn't.